pressbooks/pressbooks
# feat: Health check endpoint

Issue #2985 

This PR aims to add a health check endpoint to help with infrastructure monitoring. It adds a `/wp-json/pressbooks/v2/health-check` endpoint that tests against a couple of checks.

It will return a **200** status code with the following output if every check passes:

<img width="577" alt="image" src="https://user-images.githubusercontent.com/1761690/201950442-6edc3bc7-40d8-4c24-b56b-f4df69d7be46.png">

If one or more checks fail the status code will be **500** and the output would be something like below:

<img width="752" alt="image" src="https://user-images.githubusercontent.com/1761690/201950908-db41a9b4-4914-491f-8468-c29ec9943488.png">

This PR also removes all **GET** requests associated with `wp-json/wp/v2/users`. The response should now be a **404** with the following output:

<img width="752" alt="image" src="https://user-images.githubusercontent.com/1761690/201953450-3262ca92-c994-4e1c-b69e-51f2c547988d.png">

**How to test**

1. Add `PB_HEALTH_CHECK_TOKEN` key to your **.env** file with a random string
2. Validate `https://pressbooks.test/wp-json/pressbooks/v2/health-check` API call fails if `_token` param is missing or is different from the one in your **.env** file.
3. Validate `https://pressbooks.test/wp-json/pressbooks/v2/health-check?_token=YOUR-TOKEN-HERE` responds with a **200** status code and all checks **Ok**
4. Make sure a **500** status code is returned if at least one of the checks is not working properly
5. Make sure `wp-json/wp/v2/users` endpoint returns a 404 and cloning and exports work as expected.
